### PR TITLE
Create Parallel OIDC version of CDOGS deployment

### PR DESCRIPTION
**Pull request by jujaga**
# Description

This PR adds pre-emptive OCP4 changes, and provides templates and a plan of action for SSO to OIDC auth migration.

[SHOWCASE-1684](https://apps.nrs.gov.bc.ca/int/jira/browse/SHOWCASE-1684)

## Types of changes

New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the OpenAPI 3.0 `v*.api-spec.yaml` documentation (if appropriate)
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by bcgov-nr-csst** [comment]
SonarQube Analysis: https://sonarqube-idcqvl-tools.pathfinder.gov.bc.ca/dashboard?id=common-document-generation-service-pr-47

File: openshift/OIDC-Migration.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+Our migration strategy will have two phases. Phase 1 will be where we temporarily support two parallel deployments of the service in order to allow users to migrate on their own schedule. Phase 2 will align with the sunsetting of the sso endpoint, where we will also sunset the old endpoint, and consolidate back to only one deployment.
+
+### Phase 1
+
+Our main strategy during this period is to create a manual, parallel deploymenet of our service. This parallel deployment will accept traffic with OIDC JWTs, while we continue to support the original SSO JWTs in the original deployment. At bare minimum, we will require a parallel config map, route, service and deployment config created. A big note with these parallel resources:
```
Comment (parc-jason) [inline openshift/OIDC-Migration.md:13]: deploymenet = deployment

File: openshift/app-oidc.dc.yaml [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+apiVersion: v1
+kind: Template
+labels:
+  app.kubernetes.io/component: app
+  app.kubernetes.io/instance: "${APP_NAME}-${JOB_NAME}"
+  app.kubernetes.io/managed-by: jenkins
+  app.kubernetes.io/name: nodejs
+  app.kubernetes.io/part-of: "${APP_NAME}-${JOB_NAME}"
+  app: "${APP_NAME}-${JOB_NAME}"
+  template: "${REPO_NAME}-app-dc-template"
```
Comment (parc-jason) [inline openshift/app-oidc.dc.yaml:11]: do you want template and name to be the same as the current?  querying by template or name will bring back so and oidc objects?

File: openshift/OIDC-Migration.md (hunk 1 above)
Comment (jujaga) [reply openshift/OIDC-Migration.md:13]: fixed

File: openshift/app-oidc.dc.yaml (hunk 2 above)
Comment (jujaga) [reply openshift/app-oidc.dc.yaml:11]: Yes - normally the template label is to signify what deployment config template is managing this. We're hand-bombing this at the moment and neither our pipeline nor our cleanup instructions actually use `template` or `app.kubernetes.io/name` label selectors to identify the resources, We still select by just the `app` label in nearly all cases.

As an aside, label `app.kubernetes.io/name: nodejs` will be used in OCP4 for making the topology view have the correct nodejs icon representing the deployment config.

Comment (parc-jason) [reply openshift/app-oidc.dc.yaml:11]: ok, just don't want to accidentally remove the wrong resources.

**Review by parc-jason — APPROVED** [review]

**Pull request merged**